jrnl/plugins/dayone2_importer.py

```diff
@@ -0,0 +1,45 @@
+from datetime import datetime
+
+from jrnl.Entry import Entry
+from jrnl.plugins.json_importer import JSONImporter
+
+
+class DayOne2Importer(JSONImporter):
+
+    names = ["dayone2"]
+    extension = "json"
+
+    def __init__(self, path):
+        self.type = 'Day One 2'
+        self.path = path
+        self.keys = ['audios', 'creationDate', 'photos',
+                     'starred', 'tags', 'text', 'timeZone', 'uuid']
+        JSONImporter.__init__(self)
+        self.convert_journal()
+
+    def convert_journal(self):
+        print(self._convert())
+
+    def validate_schema(self):
+        for key in self.json['entries'][0]:
+            try:
+                assert key in self.keys
+
+            except AssertionError:
+                raise KeyError(f"{self.path} is not the expected Day One 2 format.")
+        print(f"{self.path} validated as Day One 2.")
+        return True
+
+    def parse_json(self):
+        entries = []
+        for entry in self.json["entries"]:
+            entries.append(
+                Entry(
+                    self,
+                    date=datetime.strptime(entry["creationDate"], "%Y-%m-%dT%H:%M:%SZ"),
+                    text=entry["text"],
+                    starred=entry["starred"],
```

Comment:
The UUID and tags should also be parsed and added to the entry. See https://github.com/jrnl-org/jrnl/blob/develop/jrnl/DayOneJournal.py#L62..L66 for how it is done with the DayOne classic journal.

Two questions:

1. If we import the DayOne2 journal, and find a match on the UUID in the existing journal we're importing into, what do we do? Display a diff and ask the user what to do?
2. This drops any attached photos or audio recordings. With the DayOne Classic journals, I mostly ignored this because if there was a photo, it would be in a given folder, relative to the entry, named with the UUID of the entry as the filename for the photo. So I could infer the photo location, while with DayOne2, that isn't possible. Maybe this doesn't need to be solved here, but in that case, we should raise an issue on the tracker here so it can be addressed later.

Reply:
I'm not sure I understand the purpose of saving the UUID since the date (with second resolution) is effectively a UUID.  Without knowing how DayOne2 calculates their UUIDs, introducing theirs to journals could potentially lead to duplicates in future, although that is admittedly unlikely.  Not having explicit UUIDs littered around from an exported journal would keep the .txt much cleaner as well, and reduce overhead for parsing it out when exporting.

>1. If we import the DayOne2 journal, and find a match on the UUID in the existing journal we're importing into, what do we do? Display a diff and ask the user what to do?

If either a date match or UUID match it found, I suppose it would make sense to print a diff and give a choice of choosing the old, new or aborting the import.

>2. This drops any attached photos or audio recordings. 

I'm not sure how to deal with this. My instinct is that since jrnl is a slim text-based app the most pragmatic option would be to either ignore the photos altogether.  The JSON export contains the image filename along with an MD5 hash, so this could be used to identify the associated file however the only context it has is from the associated journal entry.  I'm concerned that introducing code to manage decontextualized images would lead to feature bloat.

Reply:
> I'm not sure I understand the purpose of saving the UUID

The most obvious use would be on import. If your destination journal is a DayOne Classic journal, UUID's are going to be assigned to the entries on creation anyway, so I think it would be best to re-use the existing ones, if possible. If the same entries (possibly changed in DayOne2 on your phone) are re-imported, it makes it easy to line up what entries have changes.

I guess one of the workflows I see myself (continuing) doing is keeping a DayOne2 "journal" on my phone, exporting it to import into `jrnl`, and then exporting it out of `jrnl` via the YAML exporter to feed into Pelican, to turn everything into a static website. 

> My instinct is that since jrnl is a slim text-based app the most pragmatic option would be to ignore the photos altogether. The JSON export contains the image filename along with an MD5 hash, so this could be used to identify the associated file however the only context it has is from the associated journal entry. I'm concerned that introducing code to manage decontextualized images would lead to feature bloat.

My thought he is just to retain the data (e.g. the photo filenames, etc) as part of the Entry when imported, a little like how we try and maintain the weather information. Whether (and how) it gets saved can somewhat be handed off to the Journal and Exporter classes. I'm not envisioning any photo-handling code, although it may (down the road) make sense to add some as part of an exporter.

Reply:
```
photos: [
  {
      "exposureBiasValue":0.0,
      "height":0,
      "identifier":"9F0BB56D4EDE4BC29779ED935BBEC3F8",
      "md5":"4e2058da5f377369b4d33d994f1661ae",
      "orderInEntry":0,
      "type":"jpg",
      "width":0
   
}]
```
This is the relevant part of the DayOne2 JSON - would it be useful to append the filename to the end of the entry body, perhaps with a prefix to assist in search?

```
[2020-01-10 12:22] Entry Number One.
@awesome @great 📷9F0BB56D4EDE4BC29779ED935BBEC3F8.jpg

```
As far as the UUID goes, I'm still pretty unconvinced to be honest. They only work as intended if you're using a consistent algorithm. Since there's no way of extracting any meaningful data from the UUID or guarantee it's uniqueness in the journal, the timestamp functions as a conveniently readable UUID.

> If we import the DayOne2 journal, and find a match on the UUID in the existing journal we're importing into, what do we do? Display a diff and ask the user what to do?

I circumvented this issue by creating a new journal to hold the imported data.